src/vsr/replica.zig

```diff
@@ -1701,10 +1794,15 @@ pub fn ReplicaType(
             }
 
             if (waiting_len == 0) {
```

Comment:
Hm, given that we explicitly select prepare as `const prepare = self.primary_pipeline_pending().?;`, wouldn't this branch be dead code now?

Reply:
Aha I thought that too originally, but `waiting_len` only counts the other replicas waiting.
When `self.quorum_replication == self.replica_count`, we might be waiting for only ourself to prepare (i.e. write to the WAL).
(The assert is currently commented out because of a kludge that can be removed when state transfer is ready).